interface/patient_file/summary/create_portallogin.php

```diff
@@ -24,10 +24,10 @@
 if ($portalsite != "off" && $portalsite != "on") {
     $portalsite = "off";
 }
-
+$trustedEmail = sqlQuery("SELECT email_direct FROM `patient_data` WHERE `pid`=?", array($pid));
 $row = sqlQuery("SELECT pd.*,pao.portal_username,pao.portal_pwd,pao.portal_pwd_status FROM patient_data AS pd LEFT OUTER JOIN patient_access_" . escape_identifier($portalsite, array("on","off"), true) . "site AS pao ON pd.pid=pao.pid WHERE pd.pid=?", array($pid));
 
-function generatePassword($length = 6, $strength = 1)
+function generatePassword($length = 12, $strength = 1)
```

Comment:
for me:
look into using cryptographically secure random generator in this function

Reply:
@sjpadgett , Is your coding for this PR mostly complete? If so, I plan to make a PR based on your branch to secure the password creator function (then you can bring it into this PR).